#VRML_SIM R2020a utf8
WorldInfo {
}
Viewpoint {
  orientation -0.15320939514933118 0.939765780374178 0.30556040200865076 2.2408747792010937
  position 0.8131162054753226 0.8989322093518792 -0.5200542264593947
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
Robot {
  translation 0 -0.000229773 0
  rotation 0.011706719999931538 0.999830478516772 0.014211507160255382 -1.7672992617816705
  children [
    Transform {
      translation 0 0.0415 0
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.333333 0.666667 0.498039
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 434.013775406421
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "left wheel motor"
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation -0.04499999952643384 0.025003494964708335 -1.7953080857033946e-06
        rotation 0.9999999999999937 -4.55817258601361e-08 1.0430142552310352e-07 4.5106806396105545
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0.666667 0.498039
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "right_wheel"
        boundingObject USE WHEEL
        physics DEF PHYSICS_WHEEL Physics {
          density -1
          mass 0.05
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 186.60172670077583
        axis -1 0 0
        anchor 0 0.025 0
      }
      device [
        RotationalMotor {
          name "right wheel motor"
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0.0449999995307988 0.024998755128069497 3.7256587783392836e-06
        rotation 0.9999999999999961 -8.472150648880484e-08 -1.9282085391679312e-08 1.0956912786969089
        children [
          USE WHEEL
        ]
        name "left_wheel"
        boundingObject USE WHEEL
        physics USE PHYSICS_WHEEL
      }
    }
    DistanceSensor {
      translation 0.02 0.06 -0.04
      rotation 0 1 0 1.3090003061004252
      children [
        DEF SENSOR Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE SENSOR
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0.06 -0.04
      rotation 0 1 0 2.094399693899575
      children [
        USE SENSOR
      ]
      name "ds_right"
      boundingObject USE SENSOR
      physics Physics {
      }
    }
  ]
  boundingObject Transform {
    translation 0 0.0415 0
    children [
      USE BODY
    ]
  }
  physics Physics {
    density -1
    mass 0.5
  }
  controller "first_controller"
}
Solid {
  translation -0.17035085053675636 0.049607599875024715 -0.09234574022721219
  rotation 3.3918057482198837e-07 0.9999999999999408 -5.4188477203245455e-08 -0.0031219478799277
  children [
    Shape {
      appearance PBRAppearance {
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
}
